src/xds/client.rs

```diff
@@ -116,12 +116,18 @@ impl Client {
                     None => Err(RpcSessionError::Receive(tonic::Status::internal(
                         "Failed initial connection",
                     ))),
-                    Some(endpoint) => AggregatedDiscoveryServiceClient::connect(endpoint)
-                        .instrument(tracing::debug_span!(
-                            "AggregatedDiscoveryServiceClient::connect"
-                        ))
-                        .await
-                        .map_err(RpcSessionError::InitialConnect),
+                    Some(endpoint) => {
+                        let endpoint = Endpoint::from_shared(endpoint)
+                            .map_err(RpcSessionError::InitialConnect)?
```

Comment:
One thought I had here - if we can't create an endpoint, we could return a specific error that ends up with   - rather than the string check of "invalid url" - but wanted to check before implementing. I think we end up in the same place, since if an Endpoint can't be created, we can't ever connect. WDYT?

Reply:
That makes sense to me.